scripts/addon/build.py

```diff
@@ -0,0 +1,361 @@
+
+        if not isinstance(block["content"], list):
+            guide_strings[block_string_id] = {
+                "value": block["content"],
+                "comments": block.get("comment", block_default_comment),
+            }
+            continue
+
+        for subblock in block["content"]:
+            if not "id" in subblock:
+                exit(
+                    f"Guide {filename} does not have an id for one of the subblocks of block {block_id}"
+                )
+            if not "content" in subblock:
+                exit(
+                    f"Guide file {filename} does not have a content for subblock id {subblock['id']}"
+                )
+
+            subblock_id = subblock["id"]
+            subblock_string_id = f"guide.{guide_id}.block.{block_id}.{subblock_id}"
+            if subblock_string_id in guide_strings:
+                exit(
+                    f"Duplicate sub-block enum {subblock_string_id} when parsing {filename}"
+                )
+
+            guide_strings[subblock_string_id] = {
+                "value": subblock["content"],
+                "comments": subblock.get("comment", block_default_comment),
+            }
+
+    return guide_strings
+
+
+def write_en_language(filename, strings):
+    ts = ET.Element("TS")
+    ts.set("version", "2.1")
+    ts.set("language", "en")
+
+    context = ET.SubElement(ts, "context")
+    name = ET.SubElement(context, "name")
+
+    for key, value in strings.items():
+        message = ET.SubElement(context, "message")
+        message.set("id", key)
+
+        source = ET.SubElement(message, "source")
+        source.text = value["value"]
+
+        translation = ET.SubElement(message, "translation")
+        translation.set("type", "unfinished")
+
+        if len(value["comments"]) > 0:
+            extracomment = ET.SubElement(message, "extracomment")
+            extracomment.text = value["comments"]
+
+    with open(filename, "w", encoding="utf-8") as f:
+        f.write(ET.tostring(ts, encoding="unicode"))
+
+
+def copy_files(path, dest_path):
+    for file in os.listdir(path):
+        if file.startswith("."):
+            continue
+
+        for ext in ["qrc", "ts", "qm", "rcc"]:
+            if file.endswith(f".{ext}"):
+                exit(f"Unexpected {ext} file found: {os.path.join(path, file)}")
+
+        file_path = os.path.join(path, file)
+        if os.path.isfile(file_path):
+            shutil.copyfile(file_path, os.path.join(dest_path, file))
+            continue
+
+        if os.path.isdir(file_path):
+            dir_path = os.path.join(dest_path, file)
+            os.mkdir(dir_path)
+            copy_files(file_path, dir_path)
+
+
+def get_file_list(path, prefix):
+    file_list = []
+    for file in os.listdir(path):
+        file_path = os.path.join(path, file)
+        if (
+            os.path.isfile(file_path)
+            and not file_path.endswith(".ts")
```

Comment:
`.endswith` accepts also a tuple, so this would work

```
        if (
            os.path.isfile(file_path)
            and not file_path.endswith((".ts", ".qrc", ".rcc"))
       )
```